draft-ietf-masque-connect-udp.md

```diff
@@ -133,95 +130,35 @@ If the client has both sent and received the H3_DATAGRAM SETTINGS Parameter
 with value 1 on this connection, it SHOULD attempt to use HTTP/3 datagrams.
 This is accomplished by requesting a datagram flow identifier from the flow
 identifier allocation service {{H3DGRAM}}. That service generates an even flow
-identifier, and the client sends it to the proxy by using the unnamed element
-in a "Datagram-Flow-Id" header; see {{H3DGRAM}}. A CONNECT-UDP request with an
-odd flow identifier is malformed.
+identifier, and the client sends it to the proxy by sending an HTTP/3
+REGISTER_DATAGRAM_FLOW_ID frame with on the stream that carried the CONNECT-UDP
+request.
 
 The proxy that is creating the UDP socket to the destination responds to the
-CONNECT-UDP request with a 2xx (Successful) response, and indicates it supports
-datagram encoding by sending a "Datagram-Flow-Id" header with the same unnamed
-element from the "Datagram-Flow-Id" header it received. Once the client has
-received the "Datagram-Flow-Id" header on the successful response, it knows
+CONNECT-UDP request with a 2xx (Successful) response, and sends an HTTP/3
+REGISTER_DATAGRAM_FLOW_ID frame with the same flow ID from the
+REGISTER_DATAGRAM_FLOW_ID frame it received. Once the client has
+received the REGISTER_DATAGRAM_FLOW_ID frame and the successful response, it knows
```

Comment:
Since the HEADERS and REGISTER_DATAGRAM_FLOW_ID frames are delivered on the same stream, you have the luxury of ordering at your disposal. Since the CONNECT-UDP method is defined with `A payload within a CONNECT-UDP request message has no defined semantics`, you could state some strict frame ordering rules like "REGISTER_DATAGRAM_FLOW_ID must come after the first HEADERS on a request stream in each direction". I'm not suggesting that is necessary, just highligting something that came to mind.